src/components/DateTimePicker/DateTimePicker.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from "react";
+import throttle from "lodash.throttle";
+import { useTranslation } from "react-i18next";
+import Picker from "react-datepicker";
+
+import "react-datepicker/dist/react-datepicker.css";
+import CustomInput from './CustomInput';
+
+import en from "date-fns/locale/en-US";
+import ru from "date-fns/locale/ru";
+import hy from "date-fns/locale/hy";
+
+import {
+  DATE_FORMATS,
+  TIME_FORMATS,
+  DATE_TIME_FORMATS
+} from "../../config"
+
+
+const DateTimePicker = props => {
+
+  const LOCALES = { en, ru, hy };
+  const FORMATS = { 
+    date: DATE_FORMATS,
+    time: TIME_FORMATS,
+    datetime: DATE_TIME_FORMATS
+  }
+  const BREAKPOINT = 1000; // width less than this is considered small
+
+  const [isSmallScreen, setIsSmallScreen] = useState(false);
+
+  const calcIsSmallScreen = () => {
+    if (isSmallScreen !== (window.innerWidth < BREAKPOINT)) {
+      setIsSmallScreen(!isSmallScreen);
+    }
+  };
+
+  const calcIsSmallScreenWithThrottle = throttle(calcIsSmallScreen, 200); //throttling prevents running this code too often
+
+  useEffect(() => {
+    calcIsSmallScreen();
+    window.addEventListener("resize", calcIsSmallScreenWithThrottle);
+    return () => {
+      window.removeEventListener("resize", calcIsSmallScreenWithThrottle);
+    };
+  }, [calcIsSmallScreenWithThrottle, calcIsSmallScreen]);
+
+  const { i18n } = useTranslation();
+  const currentLanguage = i18n.language;
+
+  const {
+    date,
+    setDate,
+    type = "date",
+    customInput = CustomInput,
+    customInputProps = { icon: type.includes('date') ? 'calendar' : 'clock' },
+    onKeyDown,
+    ...rest
+  } = props;
+
+  const fixOnEnter = e => {
+    // makes sure input loses focus on enter, to avoid date parsing bugs
+    if (e.key.toLowerCase() === "enter") {
+      setTimeout(() => document.activeElement.blur());
+    }
+    if (onKeyDown) {
+      onKeyDown();
+    }
+  };
+
+  const handleDateTimeChange = changedDate => {
+    if (!type.includes("date")) {
+      return handleTimeChange(changedDate);
+    }
+    return handleDateChange(changedDate);
+  };
+
+  const handleTimeChange = newTime => {
+    const dateWithOldTime = date ? new Date(date) : new Date();      
+    const hours = newTime ? newTime.getHours() : 0;
+    const minutes = newTime ? newTime.getMinutes() : 0;
+    const dateWithUpdatedTime = new Date(
+      dateWithOldTime.setHours(hours, minutes)
+    );
+    setDate(dateWithUpdatedTime);
+  };
+
+  const handleDateChange = newDate => {
+    setDate(newDate);
+  };
+
+  return (
+    <Picker
+      selected={date}
+      onChange={handleDateTimeChange}
+      dateFormat={FORMATS[type]}
+      showTimeSelect={type.includes("time") && !isSmallScreen}
```

Comment:
Maybe we can have convenient methods like hasTime() and hasDate() so we don't duplicated this logic.

Reply:
added, not methods but rather constants